crates/pathfinder/src/observability.rs

```diff
@@ -0,0 +1,56 @@
+use futures::Future;
+use prometheus::{Encoder, TextEncoder};
+use std::net::SocketAddr;
+use std::pin::Pin;
+use warp::http::Response as HttpResponse;
+use warp::hyper::StatusCode;
+use warp::Filter;
+
+type ServerFuture = Pin<Box<dyn Future<Output = ()>>>;
+
+/// Run metrics server if `addr` is specified.
+pub fn run_server(addr: Option<SocketAddr>) -> anyhow::Result<(ServerFuture, Option<SocketAddr>)> {
+    if let Some(addr) = addr {
+        let exporter = opentelemetry_prometheus::exporter().init();
```

Comment:
Yes that's the case. Should I add a note about that in the function's doc?

Reply:
If this is required to happen before, I'd just handle the `addr: SocketAddr` case here (nested in src/obs.rs) and do the required global init in the binary `src/bin/pathfinder.rs` so that no funny stuff can happen, and add a comment that this initialization must happen before any rpc or sync things are done.